moto/awslambda/models.py

```diff
@@ -1242,7 +1246,9 @@ def put_layer_version(self, layer_version):
         self._layers[layer_version.name].attach_version(layer_version)
 
     def list_layers(self):
-        return [layer.to_dict() for layer in self._layers.values()]
+        return [
+            layer.to_dict() for layer in self._layers.values() if layer.layer_versions
```

Comment:
Just to double check..

> Actually, the current list-layer is right, even AWS returns LayerName, LayerARN and version.
> 

So we can reverse this? Or did you mean `the updated implementation for list-layer is right`?

Reply:
## What happens in AWS

Let's say we create a new Layer. The LayerStorage has a new Layer with a new LayerVersion (v1).
When we delete that Layer, we can not actually retrieve it with `list-layers` since the Layer was deleted.

However, I believe that it's still saved in LayerStorage. Why? Well, because when I create a new Layer with the same name, I will get a Layer with LayerVersion (_**v2**_). The LayerStorage keeps the Layer saved, even if it has no LayerVersions, just to keep track of the _version counter_. I think AWS soft-deletes layers.

## Moto prior to my changes

When I add a Layer and delete its LayerVersion, the call to `list-layers` raises an error since we are calling `layer.to_dict()` to a layer with no layer_versions. (the list layer_versions is empty)

## Moto after my changes

`list-layers` only returns layers with layer_versions.

Reply:
The thing is, layer.to_dict() should never be called to a Layer with no LayerVersions 😄

Reply:
Makes sense!